aspnetcore/mvc/advanced/custom-model-binding/sample/CustomModelBindingSample/Binders/AuthorEntityBinder.cs

```diff
@@ -0,0 +1,75 @@
+﻿using CustomModelBindingSample.Data;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CustomModelBindingSample.Binders
+{
+    public class AuthorEntityBinder : IModelBinder
+    {
+        private readonly AppDbContext _db;
+        public AuthorEntityBinder(AppDbContext db)
+        {
+            _db = db;
+        }
+        public Task BindModelAsync(ModelBindingContext bindingContext)
+        {
+            if (bindingContext == null)
+            {
+                throw new ArgumentNullException(nameof(bindingContext));
+            }
+
+            // specify a default argument name if none is set
+            var modelName = bindingContext.ModelName;
+            if(string.IsNullOrEmpty(modelName))
+            {
+                modelName = "authorId";
+            }
+
+            // attempt to fetch the value of the argument by name
+            var valueProviderResult = 
+                bindingContext.ValueProvider.GetValue(modelName);
+
+            // check if no matching argument exists
+            if (valueProviderResult == ValueProviderResult.None)
+            {
+                return TaskCache.CompletedTask;
+            }
+
+            bindingContext.ModelState.SetModelValue(modelName,
+                valueProviderResult);
+
+            var value = valueProviderResult.FirstValue;
+            
+            // check if the argument is null or empty
+            if (string.IsNullOrEmpty(value))
+            {
+                return TaskCache.CompletedTask;
+            }
+
+            try
+            {
+                var model = _db.Authors.Find(int.Parse(value));
```

Comment:
You should do `TryParse` here and add a model state error if the value couldn't be converted.